hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -560,24 +644,237 @@ onOpenNetworkAckSn Environment{party} openState otherParty snapshotSignature sn
+     in -- Spec: L̂   ← L̂ \ inputs(tx)
+        --       txω ← tx
+        newState (DecommitRecorded decommitTx activeUTxO)
+          <> cause (ClientEffect $ ServerOutput.DecommitRequested headId decommitUTxO)
+          -- Spec: if ŝ = S⁻.s ∧ leader(S⁻.s + 1) = i
+          --         multicast (reqSn, vˆ, S⁻.s + 1, T̂ , txω )
+          <> maybeEmitSnapshot
+ where
+  waitOnApplicableDecommit cont =
+    case mExistingDecommitTx of
+      Nothing ->
+        case applyTransactions currentSlot localUTxO [decommitTx] of
+          Right utxo' -> cont utxo'
+          Left (_, err)
+            | ttl > 0 ->
+                wait $ WaitOnNotApplicableDecommitTx decommitTx
+            | otherwise ->
+                cause . ClientEffect $
+                  ServerOutput.DecommitInvalid
+                    { headId
+                    , decommitInvalidReason =
+                        ServerOutput.DecommitTxInvalid
+                          { confirmedUTxO
+                          , decommitTx
+                          , validationError = err
+                          }
+                    }
+      Just existingDecommitTx
+        | ttl > 0 ->
+            wait $ WaitOnNotApplicableDecommitTx decommitTx
+        | otherwise ->
+            -- REVIW: cause . ClientEffect $ ServerOutput.DecommitInvalid
+            Error $ RequireFailed $ DecommitTxInFlight{decommitTx = existingDecommitTx}
+
+  maybeEmitSnapshot =
+    if isLeader parameters party nextSn
+      then cause (NetworkEffect (ReqSn version nextSn (txId <$> localTxs) (Just decommitTx)))
+      else noop
+
+  Environment{party} = env
+
+  Ledger{applyTransactions} = ledger
+
+  Snapshot{number} = getSnapshot confirmedSnapshot
+
+  nextSn = number + 1
+
+  CoordinatedHeadState{decommitTx = mExistingDecommitTx, confirmedSnapshot, localTxs, localUTxO, version} = coordinatedHeadState
+
+  confirmedUTxO = case confirmedSnapshot of
+    InitialSnapshot{initialUTxO} -> initialUTxO
+    ConfirmedSnapshot{snapshot = Snapshot{utxo}} -> utxo
+
+  OpenState
+    { headId
+    , parameters
+    , coordinatedHeadState
+    , currentSlot
+    } = openState
+
+-- ** Decrementing funds from the Head
+
+-- | Observe a decommit transaction. If the outputs of the pending decommit tx
+-- are equal to the latest confirmed UTxO to decommit, then we consider the
+-- decommit valid, and we take the funds out of the head and remove the
+-- decommit tx in flight.
+-- Finally, if the client observing happens to be the leader, then a new
+-- ReqSn is broadcasted.
+--
+-- __Transition__: 'OpenState' → 'OpenState'
+onOpenChainDecrementTx :: IsTx tx => Environment -> OpenState tx -> Outcome tx
+onOpenChainDecrementTx Environment{party} openState
+  | -- Spec: if outputs(txω) = 𝑈ω
+    -- REVIEW: should we get Uω from observation instead of local state?
+    (utxoFromTx <$> decommitTx) == utxoToDecommit =
+      -- Spec: txω ← ⊥
+      --       vˆ  ← v
+      newState DecommitFinalized
```

Comment:
Observe new version from chain (to only have + 1 in a single place -> when constructing the decrement tx)